### PR TITLE
Add first draft of notesToSelf

### PR DESCRIPTION
**Pull request by jkwok91**
TODOS
- [x] demo

### CODE REVIEW

File: workshops/notesToSelf/README.md [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+## Part I: Set-up
+
+### Creating Our Files
+
+- Open up Cloud9 workspace.
+- Create new folder in your `projects` directory, and name it `notesToSelf`.
+- Create a new file in `notesToSelf` named `index.html`
+- Create a new file in `notesToSelf` named `main.js`
+- Create a new file in `notesToSelf` named `styles.css`
+
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: Add the template for the html file

File: workshops/notesToSelf/README.md [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+}
+```
+
+Now that we've stored the note in the variable `note`, we can access the title and content.
+
+```js
+for (var i in storedNotes) {
+  var notedate = i;
+  var note = storedNotes[i];
+  noteTitle = note.title;
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: `var`

File: workshops/notesToSelf/README.md [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+  noteContent = note.content;
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: `var`

File: workshops/notesToSelf/README.md [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+
+  var thisNote = $("<div>").addClass("note");
+  var noteTitleDisplay = $("<h2>").addClass("note-title");
+  var noteDateDisplay = $("<p>").addClass("note-date");
+  var noteContentDisplay = $("<p>").addClass("note-content");
+}
+```
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: I forgot the part where you'd actually put text in these elements.

File: workshops/notesToSelf/README.md [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+We'll add these lines after accessing local storage.
+
+```js
+function postNewNote() {
+}
+
+$(".post-new-note").on("click", postNewNote);
+```
+
+Our `postNewnote()` function should:
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: `postNewNote()`

File: workshops/notesToSelf/README.md [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+There, that should look much better.
+
+### Reverse-Sorting the Notes
+
+It's a little annoying that we have to scroll all the way down to check if our newest note has been added. We should reverse-sort the notes by their date.
+
+Let's make a function to do this, at the bottom of our `main.js`:
+
+```js
+function sortObj() {
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: This is not used

File: workshops/notesToSelf/README.md [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+`index` is the variable that keeps track of the index (or position) of the date we are looking at in `sortedKeys`.
+
+And we'll add the following to make the rest of our code consistent:
+
+```js
+var keys = Object.keys(obj);
+var sortedKeys = keys.sort(compare);
+
+for (var index in sortedKeys) {
+  var i = sortedKeys[index];
```
Comment (jkwok91) [inline workshops/notesToSelf/README.md]: Clarify that we are adding this line.

**Comment by zachlatta** [comment]
Looking good so far – let me know when you want me to do a thorough review.

Let's also change the name of the workshop from `notesToSelf` to `notes_to_self` (we use snake case for every other folder in the repo).

File: workshops/notes_to_self/README.md [hunk 8]
```diff
@@ -0,0 +1,663 @@
...
+
+### Creating Our Files
+
+- Open up Cloud9 workspace.
+- Create new folder in your `projects` directory, and name it `notesToSelf`.
+- Create a new file in `notesToSelf` named `index.html`
+- Create a new file in `notesToSelf` named `main.js`
+- Create a new file in `notesToSelf` named `styles.css`
+
+In `index.htm`, add the base template, and a title "Notes to Self"
```
Comment (jkwok91) [inline workshops/notes_to_self/README.md]: typo

File: workshops/notes_to_self/README.md [hunk 9]
```diff
@@ -0,0 +1,677 @@
+# Notes to Self
+
+Short Link: [https://workshops.hackclub.com/notesToSelf/](https://workshops.hackclub.com/notesToSelf/)
```
Comment (zachlatta) [inline workshops/notes_to_self/README.md]: Needs to be updated to `notes_to_self`. The shortlink is based on the directory's name.

File: workshops/notes_to_self/README.md [hunk 10]
```diff
@@ -0,0 +1,677 @@
...
+
+Demo: [here](http://prophetorpheus.github.io/notes_to_self/)
```
Comment (zachlatta) [inline workshops/notes_to_self/README.md]: Changing `http` to `https` will work and make it more secure for the user :-).

File: workshops/notes_to_self/README.md [hunk 11]
```diff
@@ -0,0 +1,677 @@
...
+
+## Part I: Set-up
+
+### Creating Our Files
+
+- Open up Cloud9 workspace.
+- Create new folder in your `projects` directory, and name it `notesToSelf`.
+- Create a new file in `notesToSelf` named `index.html`
+- Create a new file in `notesToSelf` named `main.js`
+- Create a new file in `notesToSelf` named `styles.css`
```
Comment (zachlatta) [inline workshops/notes_to_self/README.md]: `notesToSelf` should be changed to `notes_to_self`.

**Comment by zachlatta** [comment]
Sweet, I just left a couple more comments. I'm gonna merge then fix those in a new pull request.

**Pull request merged**